File: scraper/entities/variant/Variant.py
from lib.database.createDatabaseConnection import db
from pony.orm import PrimaryKey, Required, Set, Optional


class Variant(db.Entity):
    _table_ = 'variants'
    id = PrimaryKey(int, auto=True)
    title = Required(str)
    grams = Required(float)
    available = Required(bool)
    featured_image = Optional('VariantImage')
    product = Required('Product')
    prices = Set('Price')

File: server/src/entities/variant/Variant.ts
import {
    Entity,
    PrimaryGeneratedColumn,
    Column,
    ManyToOne,
    JoinColumn,
    OneToOne,
} from 'typeorm';
import { ProductType_ } from '../product/types';
import { VariantImageType } from '../variantImage/types';
import { VariantType } from './types';

@Entity({ name: 'variants' })
export default class Variant implements VariantType {
    @PrimaryGeneratedColumn()
    id!: number;

    @Column('varchar', { length: 255, name: 'variant_title' })
    variantTitle!: string;

    @Column('double')
    grams!: number;

    @Column('bool')
    available!: boolean;

    @Column('int', { name: 'featured_image_id', nullable: true })
    featuredImageId!: number;

    @Column('int', { name: 'product_id', nullable: true })
    productId!: number;

    @OneToOne('VariantImage', 'id')
    @JoinColumn({ name: 'featured_image_id' })
    featuredImage!: VariantImageType;

    @ManyToOne('Product', 'id', { nullable: true })
    @JoinColumn({ name: 'product_id' })
    product!: ProductType_;
}
